### PR TITLE
Fix inadvertent API change

### PR DESCRIPTION
**Pull request by paulb777**
Address post-merge #9963 comment

#no-changelog

### CODE REVIEW

File: FirebaseStorage/Sources/StorageReference.swift [hunk 1]
```diff
@@ -410,15 +410,15 @@ import Foundation
    *       the current `StorageReference`.
    */
   @objc(listAllWithCompletion:)
-  open func listAll(completion: @escaping ((_: StorageListResult?, _: NSError?) -> Void)) {
+  open func listAll(completion: @escaping ((_: StorageListResult?, _: Error?) -> Void)) {
```
Comment (ncooke3) [inline FirebaseStorage/Sources/StorageReference.swift:413]: I think to have caught this, we would have needed explicit typing in the API tests: 

https://github.com/firebase/firebase-ios-sdk/blob/3d51280713332af3f023abc7e305541871c1e005/FirebaseStorage/Tests/Unit/StorageAPITests.swift#L86-L87

```swift
ref.listAll { listResult: StorageListResult?, error: Error? in 
 } 
```

**Review by ncooke3 — APPROVED** [review]
LGTM. Based on CI, it doesn't look like any tests were affected.

File: FirebaseStorage/Sources/StorageReference.swift (hunk 1 above)
Comment (ncooke3) [reply FirebaseStorage/Sources/StorageReference.swift:413]: We'd have to do this for all Swift closures to get complete coverage.

Comment (paulb777) [reply FirebaseStorage/Sources/StorageReference.swift:413]: Good point! Done for the StorageAPI tests in this PR.

**Pull request merged**